iocore/dns/DNS.cc

```diff
@@ -1553,32 +1553,56 @@ dns_process(DNSHandler *handler, HostEnt *buf, int len)
     goto Lerror;
   }
 
+  // Logs using SiteThrottled* version is helpful for noisy logs, being used here
+  // instead of print statements to help with the possible retries when a dns code occurs
   if (h->rcode != NOERROR || !h->ancount) {
     Debug("dns", "received rcode = %d", h->rcode);
     switch (h->rcode) {
     default:
-      Warning("Unknown DNS error %d for [%s]", h->rcode, e->qname);
+      SiteThrottledWarning("Unknown: DNS error %d for [%s]", h->rcode, e->qname);
       retry     = true;
       server_ok = false; // could be server problems
       goto Lerror;
-    case SERVFAIL: // recoverable error
+    case NOERROR: // NOERROR the if condition excludes this status code to avoid spamming the logs, good for testing
+      Debug("dns", "NOERROR: DNS code %d for [%s]: DNS Query completed successfully.", h->rcode, e->qname);
+      break;
+    case FORMERR: // FORMERR unrecoverable errors
+      SiteThrottledNote("FORMERR: DNS error %d for [%s]: DNS Query Format Error.", h->rcode, e->qname);
+      break;
+    case SERVFAIL: // SERVFAIL recoverable error
+      SiteThrottledNote("SERVFAIL: DNS error %d for [%s]: Server failed to complete the DNS request.", h->rcode, e->qname);
       retry = true;
       break;
-    case FORMERR: // unrecoverable errors
-    case REFUSED:
-    case NOTIMP:
-      Debug("dns", "DNS error %d for [%s]", h->rcode, e->qname);
+    case NXDOMAIN: // NXDOMAIN
+      SiteThrottledNote("NXDOMAIN: DNS error %d for [%s]: Domain name does not exist.", h->rcode, e->qname);
+      retry     = false;
+      tcp_retry = false;
+      break;
+    case NOTIMP: // NOTIMP
+      SiteThrottledNote("NOTIMP: DNS error %d for [%s]: Function not implemented.", h->rcode, e->qname);
       server_ok = false; // could be server problems
       goto Lerror;
-    case NOERROR:
-    case NXDOMAIN:
-    case 6:  // YXDOMAIN
-    case 7:  // YXRRSET
-    case 8:  // NOTAUTH
-    case 9:  // NOTAUTH
-    case 10: // NOTZONE
-      Debug("dns", "DNS error %d for [%s]", h->rcode, e->qname);
+      break;
+    case REFUSED: // REFUSED
+      SiteThrottledNote("REFUSED: DNS error %d for [%s]: The server refused to answer for the query.", h->rcode, e->qname);
       goto Lerror;
+      break;
+    case YXDOMAIN: // YXDOMAIN
+      SiteThrottledNote("YXDOMAIN: DNS error %d for [%s]: Name that should not exist, does exist.", h->rcode, e->qname);
+      break;
+    case YXRRSET: // YXRRSET
+      SiteThrottledNote("YXRRSET: DNS error %d for [%s]: RRSet that should not exist, does exist.", h->rcode, e->qname);
+      break;
+    case NXRRSET: // NXRRSET
+      SiteThrottledNote("NXRRSET: DNS error %d for [%s]: RRSet that should not exist, does not exist.", h->rcode, e->qname);
+      break;
+    case NOTAUTH: // NOTAUTH
+      SiteThrottledNote("NOTAUTH: DNS error %d for [%s]: Server not authorative for the zone.", h->rcode, e->qname);
+      break;
```

Comment:
Each of these used to `goto Lerror` before this patch. Can you please indicate whether this change in behavior in this patch is intentional for these? Judging from the description, I expected this to be a logging-only change patch, not a DNS functional change.

Reply:
please refer to my comment below.